pkg/commands/drift/statefiles/statefiles.go

```diff
@@ -0,0 +1,205 @@
+	}
+	c.directory = dirAbs
+
+	c.gitClient = git.NewGitClient(c.directory)
+	c.githubClient = github.NewClient(
+		ctx,
+		c.GitHubFlags.FlagGitHubToken,
+		github.WithRetryInitialDelay(c.RetryFlags.FlagRetryInitialDelay),
+		github.WithRetryMaxAttempts(c.RetryFlags.FlagRetryMaxAttempts),
+		github.WithRetryMaxDelay(c.RetryFlags.FlagRetryMaxDelay),
+	)
+	c.terraformParser, err = parser.NewTerraformParser(ctx, "")
+	if err != nil {
+		return fmt.Errorf("failed to create terraform parser: %w", err)
+	}
+
+	return c.Process(ctx)
+}
+
+// Process handles the main logic for the Guardian init process.
+func (c *DriftStatefilesCommand) Process(ctx context.Context) error {
+	logger := logging.FromContext(ctx).
+		Named("drift.statefiles").
+		With("github_owner", c.GitHubFlags.FlagGitHubOwner).
+		With("github_repo", c.GitHubFlags.FlagGitHubOwner)
+
+	logger.Debug("starting Guardian drift statefiles")
+	logger.Debug("finding entrypoint directories")
+
+	entrypoints, err := terraform.GetEntrypointDirectories(c.directory)
+	if err != nil {
+		return fmt.Errorf("failed to find terraform directories: %w", err)
+	}
+	entrypointBackendFiles := make([]string, 0, len(entrypoints))
+	for _, e := range entrypoints {
+		entrypointBackendFiles = append(entrypointBackendFiles, e.BackendFile)
+	}
+	logger.Debugw("terraform entrypoint directories", "entrypoint_backend_files", entrypointBackendFiles)
+
+	expectedURIs := make([]string, 0, len(entrypointBackendFiles))
+	gcsBuckets := make([]string, 0, len(entrypointBackendFiles))
+	for _, f := range entrypointBackendFiles {
+		config, _, err := terraform.ParseBackendConfig(f)
+		if err != nil {
+			return fmt.Errorf("failed to parse Terraform backend config: %w", err)
+		}
+		if config.GCSBucket == nil || *config.GCSBucket == "" {
+			return fmt.Errorf("unsupported backend type for terraform config at %s - only gcs backends are supported", f)
+		}
+		gcsBuckets = append(gcsBuckets, *config.GCSBucket)
+		expectedURIs = append(expectedURIs, fmt.Sprintf("gs://%s/%s/default.tfstate", *config.GCSBucket, *config.Prefix))
+	}
+
+	gcsBucket := c.flagGCSBucket
+	for _, bucket := range gcsBuckets {
+		if gcsBucket == "" {
+			gcsBucket = bucket
+		}
+		if bucket != gcsBucket {
+			return fmt.Errorf("found multiple definitions for gcs buckets - expected a single gcs bucket; from configs: %v; expected %s", gcsBuckets, gcsBucket)
+		}
+	}
+
+	if gcsBucket == "" {
+		return fmt.Errorf("unable to determine gcs bucket - please provide the gcsBucket flag or point at a terraform config with gcs backends")
+	}
+
+	logger.Debug("finding statefiles in gcs bucket", "gcs_bucket")
+
+	gotURIs, err := c.terraformParser.StateFileURIs(ctx, []string{gcsBucket})
+	if err != nil {
+		return fmt.Errorf("failed to determine state file URIs for gcs bucket %s: %w", gcsBucket, err)
+	}
+
+	statefilesNotInRemote := drift.DifferenceSet(Set(expectedURIs), Set(gotURIs))
+	statefilesNotInLocal := drift.DifferenceSet(Set(gotURIs), Set(expectedURIs))
```

Comment:
We've got slice functions that do this now: https://github.com/abcxyz/pkg/blob/7ad5bd201d223baaa666fec40b1d13f4f2d571a3/sets/slices.go#L143

Reply:
Done